torchok/constructor/registry.py

```diff
@@ -49,30 +70,81 @@ def get(self, key: str):
         Raises:
             KeyError: If key not in dictionary.
         """
-        if key not in self.__class_dict:
+        if key not in self.__entrypoints:
             raise KeyError(f'{key} is not in the {self.__name} registry')
-        
-        result = self.__class_dict[key]
+
+        result = self.__entrypoints[key]
 
         return result
 
-    def register_class(self, class_type: type):
-        """Register a new class.
+    def register_class(self, fn: Callable):
+        """Register a new entrypoint.
 
         Args:
-            class_type: Class to be registered.
+            fn: function to be registered.
 
         Raises:
-            TypeError: If class_type not type.
-            KeyError: If class_type already registered.
+            TypeError: If fn is not callable.
+            KeyError: If class_type is already registered.
 
         Returns:
-            Input class type.
+            Input callable object.
         """
-        if not (isinstance(class_type, type) or callable(class_type)):
-            raise TypeError(f'class_type must be class type, but got {type(class_type)}')
-        class_name = class_type.__name__
-        if class_name in self.__class_dict:
+        if not callable(fn):
+            raise TypeError(f'{fn} must be callable')
+        class_name = fn.__name__
+        if class_name in self.__entrypoints:
             raise KeyError(f'{class_name} is already registered in {self.__name}')
-        self.__class_dict[class_name] = class_type
-        return class_type
+
+        mod = sys.modules[fn.__module__]
+        module_name_split = fn.__module__.split('.')
+        module_name = module_name_split[-1] if len(module_name_split) else ''
```

Comment:
Need to explain in documentation that we save 3 dicts, and module - file name where is the model